filetests/isa/x86/prologue-epilogue.cton

```diff
@@ -228,3 +228,11 @@ ebb4:
 ; check: function %divert
 ; check: regmove v5, %rcx -> %rbx
 ; check: [Op1popq#58,%rbx]                   v15 = x86_pop.i64
+
+; Stack limit checking
+
+function %stack_limit(i64 stack_limit) {
+    ss0 = explicit_slot 168
+ebb0(v0: i64):
+    return
+}
```

Comment:
```
function %stack_limit(i64 stack_limit [%rdi], i64 fp [%rbp]) -> i64 fp [%rbp] fast {
    ss0 = explicit_slot 168, offset -184
    ss1 = incoming_arg 16, offset -16

                                ebb0(v0: i64 [%rdi], v4: i64 [%rbp]):
[RexOp1umr#8089,%rax]               v1 = copy v0
[RexOp1r_id#8081,%rax]              v2 = iadd_imm v1, 176
[RexOp1rcmp_sp#8039,%rflags]        v3 = ifcmp_sp v2
[trapif#00]                         trapif uge v3, stk_ovf
[Op1pushq#50]                       x86_push v4
[RexOp1copysp#8089]                 copy_special %rsp -> %rbp
[RexOp1adjustsp_id#d081]            adjust_sp_down_imm 176
[RexOp1adjustsp_id#8081]            adjust_sp_up_imm 176
[Op1popq#58,%rbp]                   v5 = x86_pop.i64
[Op1ret#c3]                         return v5
}

.byte 72, 137, 248, 72, 129, 192, 176, 0, 0, 0, 72, 57, 224, 114, 2, 15, 11, 85, 72, 137, 229, 72, 129, 236, 176, 0, 0, 0, 72, 129, 196, 176, 0, 0, 0, 93, 195

Disassembly:
0x0: mov rax, rdi
0x3: add rax, 0xb0
0xa: cmp rax, rsp
0xd: jb 0x11
0xf: ud2
0x11: push rbp
0x12: mov rbp, rsp
0x15: sub rsp, 0xb0
0x1c: add rsp, 0xb0
0x23: pop rbp
0x24: ret
```

does this look correct?

Reply:
Broadly yes. One issue I noticed with doing the check before the `push rbp` is that the frame size value doesn't include the 8 bytes for that push. A related issue is that if there are any `call` instructions in the body, they'll implicitly do a push of the return address, which also isn't accounted for, so you should add those to the frame size for the check.